glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/MetricsPingScheduler.kt

```diff
@@ -101,25 +71,36 @@ internal class MetricsPingScheduler(
         val millisUntilNextDueTime = getMillisecondsUntilDueTime(sendTheNextCalendarDay, now)
         Log.d(LOG_TAG, "Scheduling the 'metrics' ping in ${millisUntilNextDueTime}ms")
 
-        // Build a work request: we don't use use a `PeriodicWorkRequest`, which
-        // is more suitable for this task, because of the inherent drifting. See
-        // https://developer.android.com/reference/androidx/work/PeriodicWorkRequest.html
-        // for more details.
-        val workRequest = OneTimeWorkRequestBuilder<MetricsPingWorker>()
-            .addTag(MetricsPingWorker.TAG)
-            .setInitialDelay(millisUntilNextDueTime, AndroidTimeUnit.MILLISECONDS)
-            .build()
+        // Cancel any existing scheduled work. Does not actually cancel a
+        // currently-running task.
+        timer?.cancel()
```

Comment:
nit: doesn't really change much but, since we're defining a `cancel()` function, would it make sense to use it here as well?